File: nodes/array-map/array-map.process.py
import math

async def process(inputs, settings, config, nodeConfig):
  array = inputs.get("array", [])
  if not isinstance(array, list):
    array = [array]
    
  property_name = inputs.get("property", "")
  
  def extract_property(item):
    if isinstance(item, dict):
      return item.get(property_name)
    return None
    
  return {
    "array": list(map(extract_property, array))
  } 

File: sdks/nodejs/nodes/array-map/array-map.process.js
export default async ({inputs, settings, config}) => {

  const array = Array.isArray(inputs.array) ? inputs.array : [inputs.array];
  const property = inputs.property || "";

  const transformed = array.map(item => {
    if (!item || typeof item !== 'object') {
      return null;
    }
    return property in item ? item[property] : null;
  });

  return {
    array: transformed
  };
}; 